dgil/node-chat
# Basic features

Test started at 12:05. 
Test ended at 14:20.

The code is quite ugly but "it works". It doesn't have any validation or tests 😢 

### How to test it

#### Setup
1. Run `node main.js`
1. Go to [localhost:3000](http://localhost:3000)
1. Type `/nick {the nickname you wish}`
1. Open another tab / browser [localhost:3000](http://localhost:3000)
1. Type `/nick {the nickname you wish}` (this one should be different than the step 3)

At this point you have the two _users_ connected to the server using websockets.

#### Testing the commands
1. Type something, e.g. `hello world` and you should see it on green in the tab where you send it and in grey border in the other user's tab
1. Try again in the other browser and see that the behaviour is as described in the previous step
1. Type `/think some message`: you should see in both tabs the text's color in grey instead of black
1. Type `/oops` and you should see the last message of the user disappear in both tabs

### Know issues

* There is neither validation of user input nor commands
* The user cases I've tested are quite limited, so if you try different commands the behaviour is unknown
* There are no tests
